### PR TITLE
Windows packages plugin - Get packages from registry

### PR DESCRIPTION
**Pull request by sh9189**
Windows packages plugin uses Win32_Product WMI class to get list of installed packages. This causes  reconfiguration of all installed windows packages in win2008r2/win2012r2 as per https://support.microsoft.com/en-us/kb/974524, spewing windows event log with 'Package reconfigured messages'. The PR removes the query to Win32_Product WMI class  and uses windows registry to get list of all installed packages.
cc @johnbellone

### CODE REVIEW

**Comment by tas50** [comment]
@sh9189 Any chance you can benchmark this?  We previously disable the plugin due to the slow performance using WMI.  I'm curious if this would allow us to enable this out of the box now.

**Comment by sh9189** [comment]
Sure, will work on getting some benchmarks for comparison with the previous version.

**Comment by sh9189** [comment]
@tas50 I did some simple tests on my local PC running windows 7. These are results I see,

                                                           | Run 1 | Run 2 | Run 3 | Run 4 | Run 5 
 --------------------------------------------   | -------- | ------- | -------- | -------- | -------- 
Use WMI (without packages plugin) |  23.070s | 21.623s | 21.685s | 21.882s | 23.454s                 
Use WMI (with packages plugin)      | 30.919s | 30.447s | 30.608s | 30.608s | 30.892s
Use registry (without packages plugin) | 23.927s | 23.693s | 23.329s | 23.450s | 23.924s
Use registry (with packages plugin) | 24.480s | 24.504s | 24.429ss | 24.308s | 24.689s

When using WMI to query packages, the packages plugin adds almost 7-8 seconds, but using windows registry to query packages, the packages plugin adds only 0.5-1 second.

I have a pushed a commit to enable packages plugin by default to see how it performs on appveyor.

**Comment by sh9189** [comment]
On appveyor, I just used start time and end time for `bundle exec ohai` command to get time taken for ohai. These are results I see,

                                                           | Ruby 2.0 | Ruby 2.1| Ruby 2.2 
 --------------------------------------------   | -------- | ------- | -------- 
Use registry (without packages plugin) | 16:20 - 16:38 = 18 seconds |16:09 - 16:25 = 16 seconds | 16:14 - 16:30 = 16 seconds
Use registry (with packages plugin) |16:22 - 16:40 = 18 seconds | 01:57 - 02:14 = 17 seconds | 16:10 - 16:26 = 16 seconds

The packages plugin adds almost little to no overhead on appveyor when reading from windows reqistry.

File: lib/ohai/plugins/packages.rb [hunk 1]
```diff
@@ -22,63 +22,78 @@
...
   depends "platform_family"
 
+  WINDOWS_ATTRIBUTE_ALIASES = {
+    "DisplayVersion" => "version",
+    "Publisher" => "publisher",
+    "InstallDate" => "installdate"
+  }
+
   collect_data(:linux) do
-    if configuration(:enabled)
```
Comment (mcquin) [inline lib/ohai/plugins/packages.rb]: It looks like this plugin was disabled by default [here](https://github.com/chef/ohai/pull/717). Are you intentionally reverting that?

Comment (mcquin) [inline lib/ohai/plugins/packages.rb]: OK I think I have the proper background on why this plugin was disabled. It seems safe to enable it again. :)

**Comment by mcquin** [comment]
Code looks great! Can we get some reviews from @chef/client-windows ?

**Comment by tas50** [comment]
:+1:

**Comment by ksubrama** [comment]
:+1:

**Comment by Roviluca** [comment]
+1
Are you going to merge?

**Pull request merged**